### ISSUE TITLE
Intended config not generated while using arista avd version v3.8.6

### ISSUE THREAD

**Issue by duddis1** [aristanetworks/ansible-avd#3154]
### Issue Summary

HI Team,

We are getting below error while generating intended config using arista avd version v3.8.6.

This has worked till yesterday, do you know any validation has been added on top of this image.

### How do you run AVD ?

None

### Relevant log output

```shell
Error message from log:
TASK [arista.avd.eos_cli_config_gen : Generate eos intended configuration] ****************************************************************************************************************************************
task path: /home/avd/.ansible/collections/ansible_collections/arista/avd/roles/eos_cli_config_gen/tasks/main.yml:38
The full traceback is:
Traceback (most recent call last):
  File "/home/avd/.ansible/collections/ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschema.py", line 53, in load_schema
    self._validator = AvdValidator(schema)
TypeError: 'NoneType' object is not callable

 

The above exception was the direct cause of the following exception:

 

Traceback (most recent call last):
  File "/home/avd/.ansible/collections/ansible_collections/arista/avd/plugins/action/validate_and_template.py", line 66, in run
    self.avd_schema = AvdSchema(schema)
  File "/home/avd/.ansible/collections/ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschema.py", line 40, in __init__
    self.load_schema(schema)
  File "/home/avd/.ansible/collections/ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschema.py", line 57, in load_schema
    raise AristaAvdError("An error occured during creation of the validator") from e
ansible_collections.arista.avd.plugins.plugin_utils.errors.errors.AristaAvdError: An error occured during creation of the validator
fatal: [MBH-CAMP0X -> localhost]: FAILED! => {
    "changed": false,
    "msg": "Invalid Schema supplied to the 'arista.avd.validate_and_template' plugin"
}
The full traceback is:
Traceback (most recent call last):
  File "/home/avd/.ansible/collections/ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschema.py", line 53, in load_schema
    self._validator = AvdValidator(schema)
TypeError: 'NoneType' object is not callable

 

The above exception was the direct cause of the following exception:

 

Traceback (most recent call last):
  File "/home/avd/.ansible/collections/ansible_collections/arista/avd/plugins/action/validate_and_template.py", line 66, in run
    self.avd_schema = AvdSchema(schema)
  File "/home/avd/.ansible/collections/ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschema.py", line 40, in __init__
    self.load_schema(schema)
  File "/home/avd/.ansible/collections/ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschema.py", line 57, in load_schema
    raise AristaAvdError("An error occured during creation of the validator") from e
ansible_collections.arista.avd.plugins.plugin_utils.errors.errors.AristaAvdError: An error occured during creation of the validator
fatal: [MBH-CAMP0W -> localhost]: FAILED! => {
    "changed": false,
    "msg": "Invalid Schema supplied to the 'arista.avd.validate_and_template' plugin"
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

**Comment by gmuloc** [comment]
Hi,

thanks for raising this issue. If I understand correctly everything was working fine until yesterday and this is now failing but you are still using the collection 3.8.6.

There is a path in the code of 3.8.6 that can trigger AvdValidator to be `None` like your stacktrace is showing:
https://github.com/aristanetworks/ansible-avd/blob/v3.8.6/ansible_collections/arista/avd/plugins/plugin_utils/schema/avdvalidator.py#L171

This would indicate the loading of either `jsonschema` or `deepmerge` failed in in your environment.

By any chance, have you installed some new packages that could have upgraded the versions of one of these packages you had? Can you please verify using `pip freeze` the versions of `jsonschema` and `deepmerge` you are currently running in your environment?

To unlock your current setup you can probably run the following command (more explanation below):
```
pip install "jsonschema>=4.5.1,<4.18"
```

---

If look at the requirements for AVD for 3.8.6 (https://github.com/aristanetworks/ansible-avd/blob/v3.8.6/ansible_collections/arista/avd/requirements.txt) you can see that we have no upper bound on `jsonschema` (because none was needed at the time 3.8.6 was released).

However in our most recent releases (e.g. 4.3.0: https://github.com/aristanetworks/ansible-avd/blob/v4.3.0/ansible_collections/arista/avd/requirements.txt#L6) you will see that we now need an upper bound on jsonschema as more recent versions are introducing breaking change with our code.


Let us know how it goes

**Comment by duddis1** [comment]
Thanks for this suggestion, this solution worked fine, and we can generate the config with v3.8.6

**Comment by gmuloc** [comment]
Hello @duddis1 FYI we just published 3.8.7 which has a fix to prevent this issue to reoccuring: https://github.com/aristanetworks/ansible-avd/releases/tag/v3.8.7
